src/main/java/seedu/address/logic/commands/AddShortcutCommand.java

```diff
@@ -0,0 +1,78 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_COMMAND_WORD;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_SHORTCUT;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+
+/**
+ * Adds a shortcut to the address book.
+ */
+public class AddShortcutCommand extends Command {
+    public static final String COMMAND_WORD = "addsc";
+    public static final String MESSAGE_SUCCESS = "New shortcut added: %1$s";
+    public static final String MESSAGE_REPLACED = "Old shortcut %1$s was removed as a result.";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Registers the shortcut to the valid command keyword for future use.\n"
+            + "Parameters: "
+            + PREFIX_SHORTCUT + "SHORTCUT "
+            + PREFIX_COMMAND_WORD + "KEYWORD\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_SHORTCUT + "del "
+            + PREFIX_COMMAND_WORD + DeleteCommand.COMMAND_WORD;
+    private final ShortcutAlias shortcutAlias;
+    private final CommandWord command;
+
+    /**
+     * Creates an AddShortcutCommand to add the specified shortcut mapping.
+     */
+    public AddShortcutCommand(ShortcutAlias shortcutAlias, CommandWord commandWord) {
+        requireAllNonNull(shortcutAlias, commandWord);
+        this.shortcutAlias = shortcutAlias;
+        this.command = commandWord;
+    }
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        String feedback = model.getShortcutSettings().registerShortcut(shortcutAlias, command);
+        if (feedback == null) {
```

Comment:
Great use of feedback LinkedHashMap return value as feedback for override scenario!